iothub_client/src/uamqp_messaging.c

```diff
@@ -24,6 +24,10 @@
 
 #define MESSAGE_ID_MAX_SIZE 128
 
+#define AMQP_DIAGNOSTIC_ID_KEY "Diagnostic-Id"
+#define AMQP_DIAGNOSTIC_CONTEXT_KEY "Correlation-Context"
```

Comment:
"Correlation-Context" is too generic. Please make sure the name clearly links to E2E diagnostics.

Reply:
please see my comment below.